### PR TITLE
Email campaign incidents info fix

### PR DESCRIPTION
**Pull request by iyeshaya**
## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Description
Fixed display fields bug

## Minimum version of Cortex XSOAR
- [x] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 1]
```diff
@@ -45,52 +46,67 @@ def update_incident_with_required_keys(incidents, required_keys):
...
         :type incident: ``dict``
         :param incident: the incident to get the value from
 
         :type key: ``str``
         :param key: the key in dict
 
-        :rtype: ``str``
-        :return the value form dict
+        :rtype: ``None``
+        :return None
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:61]: this seems like a mistake.

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 2]
```diff
@@ -45,52 +46,67 @@ def update_incident_with_required_keys(incidents, required_keys):
...
     """
+
     if incidents:
-        incidents_info = [
-            {key: get_incident_val(incident, key) for key in INCIDENTS_HEADER} for incident in incidents
-        ]
+        if not fields_to_display:
+            headers = HEADER_SORTED
+        else:
+            headers = [header for header in HEADER_SORTED if fields_to_display]
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:100]: wouldn't this just give me the `HEADER_SORTED` list if `fields_to_display` exists?
I think you meant:
`headers = [header for header in HEADER_SORTED if header in fields_to_display]`

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py [hunk 3]
```diff
@@ -45,52 +46,67 @@ def update_incident_with_required_keys(incidents, required_keys):
...
+            headers += list(set(fields_to_display) - set(HEADER_SORTED))
```
Comment (roysagi) [inline Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py:101]: add a dev comment about what you are doing here as it looks funny though the idea if I get it is good - adding the rest of the field to display after the preset fields.

**Review by roysagi — CHANGES_REQUESTED** [review]

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py (hunk 1 above)
Comment (iyeshaya) [reply Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py]: Fixed

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py (hunk 2 above)
Comment (iyeshaya) [reply Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py]: This section has been changed.

File: Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py (hunk 3 above)
Comment (iyeshaya) [reply Packs/Campaign/Scripts/GetCampaignIncidentsInfo/GetCampaignIncidentsInfo.py]: This section has been changed.

**Review by roysagi — APPROVED** [review]

**Pull request merged**